pkg/k8s/watchers/pod.go

```diff
@@ -719,6 +719,11 @@ func (k *K8sWatcher) deleteHostPortMapping(pod *slim_corev1.Pod, podIPs []string
 	}
 
 	for _, dpSvc := range svcs {
+		svcs := k.svcManager.GetDeepCopyServices()
+		svc := findServiceByL3n4Addr(dpSvc.Frontend.L3n4Addr, svcs)
```

Comment:
You can skip all this looping (and the full deepcopy above) by adding this method to the ServiceManager:

Actually, you can skip the looping entirely. Just add this method to the ServiceManager, and we skip the O(N^2) complexity:

```go
func (s *Service) GetDeepCopyServiceByFrontend(frontend lb.L3n4Addr) (*lb.SVC, bool) {
	s.Lock()
	defer s.Unlock()

	if svc, found := s.svcByHash[frontend.Hash()]; found {
		return svc.deepCopyToLBSVC(), true
	}

	return nil, false
}
```

Then you won't need `findServiceByL3nL4Addr` or the full `GetDeepCopyServices`

Reply:
Done!